components/centraldashboard/public/components/main-page.js

```diff
@@ -141,8 +141,8 @@ export class MainPage extends utilitiesMixin(PolymerElement) {
      * Resync the app with environment information
      */
     async resyncApp() {
-        this.$.envInfo.generateRequest();
-        await this.sleep(500);
+        await this.$.envInfo.generateRequest().completes;
+        await this.sleep(100);
```

Comment:
Is this additional wait necessary now that you're blocking on the completion of the API call?